Add flags to conferences

HedicGuibert opened the pull request:
This PR addresses #3.

It adds a country and city fields to the Conference entity rather than a location field, and it displays them with flag in the nextConference adminView.

How it looks on the frontend :
![Frontend flag](https://user-images.githubusercontent.com/71645693/102369746-961fcc80-3fbc-11eb-8e25-964f7f54486d.png)

How it looks in the admin :
![Flags View](https://user-images.githubusercontent.com/71645693/102369867-bb143f80-3fbc-11eb-9257-bd2be6e77e54.png)

How to add a new country :
![Save Conf](https://user-images.githubusercontent.com/71645693/102369910-c7989800-3fbc-11eb-8de4-442bb74fc29b.png)

What it looks like now:
![Flags View2](https://user-images.githubusercontent.com/71645693/102369935-cebfa600-3fbc-11eb-9f7e-cd40c7134691.png)

welcoMattic commented on src/Entity/Conference.php:
```diff
@@ -60,11 +60,6 @@ class Conference
-     * @ORM\Column(name="location", type="string", length=255)
-     */
-    private $location;
-
```
Be careful here, we will lost data stored in this field. Be sure that your migration will move data from Location to City or Country fields

welcoMattic commented on migrations/Version20201216112814.php:
```diff
@@ -0,0 +1,44 @@
+        $this->abortIf('postgresql' !== $this->connection->getDatabasePlatform()->getName(), 'Migration can only be executed safely on \'postgresql\'.');
+
+        $this->addSql('DROP INDEX uniq_911533c8d1b862b8');
+        $this->addSql('ALTER TABLE conference DROP hash');
```
Why `hash` column is dropped here? It looks like an mistake

HedicGuibert replied:
I agree, I don't know what it's doing here, I saw it and thought it may have been done previously and not added to a migration so I let it here, but I can drop it :+1:

lyrixx replied:
I merged another PR where he fixed this stuff :)

welcoMattic commented on migrations/Version20201216112814.php:
```diff
@@ -0,0 +1,44 @@
+
+        $this->addSql('DROP INDEX uniq_911533c8d1b862b8');
+        $this->addSql('ALTER TABLE conference DROP hash');
+        $this->addSql('ALTER TABLE conference RENAME COLUMN location TO city');
```
Nice :ok_hand:

welcoMattic commented:
@HedicGuibert We want to display the flag on the frontend side too

lyrixx commented on migrations/Version20201216112814.php:
```diff
@@ -0,0 +1,44 @@
+{
+    public function up(Schema $schema): void
+    {
+        $this->abortIf('postgresql' !== $this->connection->getDatabasePlatform()->getName(), 'Migration can only be executed safely on \'postgresql\'.');
```
Should be removed (I hate theses checks :))

lyrixx commented on migrations/Version20201216112814.php:
```diff
@@ -0,0 +1,44 @@
+        $this->addSql('DROP INDEX uniq_911533c8d1b862b8');
+        $this->addSql('ALTER TABLE conference DROP hash');
+        $this->addSql('ALTER TABLE conference RENAME COLUMN location TO city');
+        $this->addSql('ALTER TABLE conference ADD country VARCHAR(255) DEFAULT \'Undefined\' NOT NULL');
```
I don't like this "Undefined". Let's use `null` is done for that

HedicGuibert replied:
I could but the field is not nullable. Should I set it to Nullable then ? Or maybe this will work work anyway ?

lyrixx replied:
if the country could be missing, so yes the field must be nullable

lyrixx commented on migrations/Version20201216112814.php:
```diff
@@ -0,0 +1,44 @@
+        $this->addSql('ALTER TABLE conference DROP hash');
+        $this->addSql('ALTER TABLE conference RENAME COLUMN location TO city');
+        $this->addSql('ALTER TABLE conference ADD country VARCHAR(255) DEFAULT \'Undefined\' NOT NULL');
+        $this->addSql('ALTER TABLE users ALTER email TYPE VARCHAR(255)');
```
This is suspect

lyrixx commented on migrations/Version20201216112814.php:
```diff
@@ -0,0 +1,44 @@
+
+    public function down(Schema $schema): void
+    {
+        $this->abortIf('postgresql' !== $this->connection->getDatabasePlatform()->getName(), 'Migration can only be executed safely on \'postgresql\'.');
+
+        $this->addSql('CREATE SCHEMA public');
```
To remove

lyrixx commented on src/Entity/Conference.php:
```diff
@@ -116,6 +111,16 @@ class Conference
+    /**
+     * @ORM\Column(type="string", length=255)
+     */
+    private $country;
```
```suggestion
    private ?string $country = null;
```

same after. We are using PHP 7.4, let's use it :)

lyrixx reviewed (commented):
I don't see the code that display flags
+ I don't see where you set the city / location
+ Could you add a screenshot?

lyrixx commented:
@HedicGuibert Could you rebase to get ride of not-related SQL diff? thanks

HedicGuibert commented:
I updated the description and made the requested changes.
@welcoMattic You may want to look closely at the NewConferenceEvent, since I added a new attachment that may not be right.

welcoMattic commented:
@HedicGuibert Looks good, in a Slack block you can add many attachments, no worries about adding one for City ;)

lyrixx commented:
Hi, I don't understand how this PR can work.
Basically, you renamed `setLocation` to `setCity` and `setCountry`.
I saw you change things in the fixtures and the admin. This is nice.

But I did not see any change in the frontend, nor in the fetcher, etc
=> How could it continue to work ?
=> Do the command to fetch conference continue to work ?
You need to search in the code for every single usage of get/setLocation() and update them.

lyrixx commented on config/services.yaml:
```diff
@@ -49,3 +49,18 @@ services:
         arguments:
             $allowedGitHubOrganization: '%env(APP_ALLOWED_GITHUB_ORGANIZATION)%'
+
+    client:
```
this name is too generic / vague. Please use something like `http.guzzle.client`

lyrixx replied:
Nevermind: use the classname for the ie, same after

lyrixx reviewed (commented):
Except one little comment, this seems good !

lyrixx commented:
I'll wait for PHP before merging this one

lyrixx commented:
Thanks, good job 👍🏼
